components/src/NavBar/DesktopMenu.js

```diff
@@ -6,29 +6,73 @@ import MenuTrigger from "./MenuTrigger";
 import MenuLink from "./MenuLink";
 import theme from "../theme";
 
-const isTrigger = menuItem => (menuItem.items);
+const ApplyMenuLinkStyles = styled.div({
+  "& *": {
+    display: "inline-flex",
+    color: theme.colors.white,
+    textDecoration: "none",
+    border: "none",
+    backgroundColor: "transparent",
+    justifyContent: "center",
+    alignItems: "center",
+    verticalAlign: "middle",
+    lineHeight: theme.lineHeights.base,
+    transition: ".2s",
+    fontSize: `${theme.fontSizes.medium}`,
+    padding: `${theme.space.x1} ${theme.space.x2}`,
+    borderRadius: theme.radii.medium,
+    "&:hover, &:focus": {
+      outline: "none",
+      color: theme.colors.lightBlue,
+      backgroundColor: theme.colors.black,
+      cursor: "pointer",
+    },
+    "&:disabled": {
+      opacity: ".5",
+    },
+  },
+});
+
+const renderMenuTrigger = menuItem => (
+  <div key={ menuItem.name }>
+    <MenuTrigger name={ menuItem.name } menuData={ menuItem.items } />
+  </div>
+);
+
+const renderMenuLink = menuItem => (
+  <div key={ menuItem.name }>
+    <MenuLink href={ menuItem.href }>
+      {menuItem.name}
+    </MenuLink>
+  </div>
+);
+
+const renderCustom = menuItem => (
+  <ApplyMenuLinkStyles key={ menuItem.name }>
+    {menuItem.render()}
+  </ApplyMenuLinkStyles>
+);
+
+const getRenderFunction = menuItem => {
+  if (menuItem.items) {
+    return renderMenuTrigger;
+  } else if (menuItem.href) {
+    return renderMenuLink;
+  } else if (menuItem.render) {
+    return renderCustom;
+  } else {
+    return (() => (null));
+  }
+};
 
 const BaseDesktopMenu = ({
   menuData,
   ...props
 }) => (
   <Flex { ...props }>
     {menuData.map(menuItem => {
```

Comment:
Could be shortened to:
`menuData.map(menuItem => getRenderFunction(menuItem)(menuItem))`

which could then be shortened to:
`menuData.map(menuItem => renderMenuItem(menuItem))`

which could be shortened to:
`menuData.map(renderMenuItem)`

Thoughts?

Reply:
I cleaned up this instance in DesktopMenu and in MenuTrigger and SubMenuTrigger. The mapping in MobileMenu.js is a bit more complex on how the render function is determined and what is passed to it so I left that one a bit more verbose to keep it readable.